docs/apis/common/index.md

```diff
@@ -1350,3 +1350,211 @@ You may specify program arguments before the job is executed. The plan visualiza
 the execution plan before executing the Flink job.
 
 {% top %}
+
+Metrics
+-------------------
+
+Flink exposes a metric system that allows users gather and expose metrics to external systems.
+
+### Registering metrics
+
+You can access the metric system from any user function that extends [RichFunction]({{ site.baseurl }}/apis/common/index.html#rich-functions) by calling getRuntimeContext().getMetricGroup().
+This method returns a MetricGroup object on which you can create and register new metrics.
+
+If you want to count the number of records your user function has received you could use a Counter like this:
+
+{% highlight java %}
+
+public class MyMapper extends RichMapFunction<String, Integer> {
+  private Counter counter;
+
+  @Override
+  public void open(Configuration config) {
+    // create and register a counter
+    this.counter = getRuntimeContext().getMetricGroup().counter("myCounter");
+    ...
+  }
+
+  @public Integer map(String value) throws Exception {
+    // increment counter
+    this.counter.inc();
+    ...
+  }
+}
+
+{% endhighlight %}
+
+### Metric types
+
+Flink supports Counters, Gauges and Histograms.
+
+A Counter is used to count something. The current value can be in- or decremented using `inc([long n])` or `dec([long n])`.
+You can create and registers a Counter by calling `counter(String name)` on a MetricGroup. Alternatively, you can pass an instance of your own Counter implementation along with the name.
+
+A Gauge provides a value of any type on demand. In order to use a Gauge you must first create a class that implements the `org.apache.flink.metrics.Gauge` interface.
+There is not restriction for the type of the returned value.
+You can register a gauge by calling `gauge(String name, Gauge gauge)` on a MetricGroup.
+
+A Histogram measure the distribution of long values.
+You can register one by calling histogram(String name, Histogram histogram) on a MetricGroup.
+
+Flink only provides an interface for Histograms, but offers a Wrapper that allows usage of Codahale/DropWizard Histograms. (org.apache.flink.dropwizard.metrics.DropWizardHistogramWrapper)
```

Comment:
I think it would be better to present the different metric types either as a descriptive list similar to the description environment of latex or by using sub headings. It could also be more descriptive to give for each metric type a short code snippet how to register and interacting with it. That would explain their interface which is important for the user.